From 36838fa39d58cb0819084d76fc0ffece9c3f1c58 Mon Sep 17 00:00:00 2001
From: udeved <artoo@manjaro.org>
Date: Wed, 23 Dec 2015 21:49:05 +0100
Subject: [PATCH] lxqt-sudo support

add a method to retrieve lxqt-sudo used with su backend

Signed-off-by: udeved <artoo@manjaro.org>
---
 src/wmhelper.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/wmhelper.h b/src/wmhelper.h
index de112e0..b92fd36 100644
--- a/src/wmhelper.h
+++ b/src/wmhelper.h
@@ -112,6 +112,7 @@ public:
   static QString getKDESUCommand();
   static QString getTDESUCommand();
   static QString getGKSUCommand();
+  static QString getLXQTSUCommand();
   static QString getXFCEEditor();
   static QString getSUCommand();
   static void openFile(const QString& fileName); //fileName is Path + Name
--

---
 src/wmhelper.cpp | 15 ++++++++++++++-
 src/wmhelper.h   |  2 ++
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/src/wmhelper.cpp b/src/wmhelper.cpp
index b3f21b9..a373b8b 100644
--- a/src/wmhelper.cpp
+++ b/src/wmhelper.cpp
@@ -293,6 +293,17 @@ QString WMHelper::getGKSUCommand(){
 }

 /*
+ * Retrieves the LXQTSU command...
+ */
+QString WMHelper::getLXQTSUCommand(){
+  QString result = ctn_LXQTSU;
+
+  result += " -s ";
+
+  return result;
+}
+
+/*
  * The generic SU get method. It retrieves the SU you have installed in your system!
  */
 QString WMHelper::getSUCommand(){
@@ -316,7 +327,9 @@ QString WMHelper::getSUCommand(){
   else if (UnixCommand::hasTheExecutable(ctn_TDESU)){
     result = getTDESUCommand();
   }
-
+  else if (UnixCommand::hasTheExecutable(ctn_LXQTSU)){
+    result = getLXQTSUCommand();
+  }
   return result;
 }

diff --git a/src/wmhelper.h b/src/wmhelper.h
index 93d793f..de112e0 100644
--- a/src/wmhelper.h
+++ b/src/wmhelper.h
@@ -28,6 +28,8 @@ const QString ctn_ROOT_SH("/bin/sh -c ");

 const QString ctn_KDESU("kdesu");

+const QString ctn_LXQTSU("lxqt-sudo");
+
 const QString ctn_KDE_DESKTOP("kwin");
 const QString ctn_KDE_X11_DESKTOP("kwin_x11");
 const QString ctn_KDE_WAYLAND_DESKTOP("kwin_wayland");
--
2.6.4

